### PR TITLE
fix: [PIE-6404]: Do not call useGetSettingValue API for Github Webhook Trigger Authentication if NG_SETTINGS is disabled

### PR DESCRIPTION
**Pull request by pankajsingh-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by pankajsingh-harness** [comment]
retrigger titlecheck

**Review by AshwinBhatkal-Harness — CHANGES_REQUESTED** [review]
We can pass lazy as true if NG_SETTINGS is on and handle all of this. We don't need the extra states. Also, we need to show error on error

File: src/modules/72-triggers/components/Triggers/WebhookTrigger/useIsGithubWebhookAuthenticationEnabled.tsx [hunk 1]
```diff
@@ -1,37 +1,46 @@
-/*
```
Comment (AshwinBhatkal-Harness) [inline src/modules/72-triggers/components/Triggers/WebhookTrigger/useIsGithubWebhookAuthenticationEnabled.tsx:1]: This shouldn't have been removed. Did we skip the pre commit check?

Comment (pankajsingh-harness) [reply src/modules/72-triggers/components/Triggers/WebhookTrigger/useIsGithubWebhookAuthenticationEnabled.tsx:1]: It was taking time for that. So skipped that. Let me resolve this

**Comment by pankajsingh-harness** [comment]
retrigger cypress-cd

File: src/modules/72-triggers/components/Triggers/WebhookTrigger/useIsGithubWebhookAuthenticationEnabled.tsx [hunk 2]
```diff
@@ -5,33 +5,49 @@
...
+  useEffect(() => {
+    if (projectSettingDataError) {
+      showError(getErrorMessage(projectSettingDataError))
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [projectSettingDataError])
+
+  const isGithubWebhookAuthenticationEnabled = useMemo(
+    () => projectSettingData?.data?.value === 'true',
```
Comment (AshwinBhatkal-Harness) [inline src/modules/72-triggers/components/Triggers/WebhookTrigger/useIsGithubWebhookAuthenticationEnabled.tsx:48]: Last question 😛 Is 'true' coming as a string from the BE?

**Review by AshwinBhatkal-Harness — APPROVED** [review]

File: src/modules/72-triggers/components/Triggers/WebhookTrigger/useIsGithubWebhookAuthenticationEnabled.tsx (hunk 2 above)
Comment (pankajsingh-harness) [reply src/modules/72-triggers/components/Triggers/WebhookTrigger/useIsGithubWebhookAuthenticationEnabled.tsx:48]: Yes. 
export interface SettingValueResponseDTO {
  value: string
  valueType: 'String' | 'Boolean' | 'Number'
}

**Review by Tamaraius — APPROVED** [review]

**Pull request merged**